ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java

```diff
@@ -333,9 +390,14 @@ void addToIndex(IndexEntry entry, FileSpan fileSpan) throws StoreException {
       IndexSegment info = new IndexSegment(dataDir, entry.getValue().getOffset(), factory, entry.getKey().sizeInBytes(),
           entry.getValue().getBytes().capacity(), config, metrics);
       info.addEntry(entry, fileSpan.getEndOffset());
-      indexes.put(info.getStartOffset(), info);
+      indexesChangeLock.lock();
```

Comment:
Discussion is the same as [this](https://github.com/linkedin/ambry/pull/543#discussion_r102541125).